src/wizards/generateProject/generationWizard.ts

```diff
@@ -167,12 +167,13 @@ async function getTargetDirectory(projectName: string) {
   const OPTION_OVERWRITE = 'Overwrite';
   const OPTION_CHOOSE_NEW_DIR = 'Choose new directory';
 
-  let directory: Uri|undefined = await showOpenFolderDialog({ openLabel: LABEL_CHOOSE_FOLDER });
+  const defaultDirectoryUri = workspace.workspaceFolders ? workspace.workspaceFolders[0].uri : undefined;
```

Comment:
Could this use some kind of guard? Under Theia (with this change), if I attempt to run the wizard when no workspace has been selected, I get an exception.

Maybe something like (array && array.length) is needed.

Reply:
Should probably be `(workspaceFolders && workspaceFolders.length > 0) ?....`

Reply:
Thank you @rgrunber for checking it.
I believe no additional guard is needed that would be a Theia-specific one.
`workspace.workspaceFolders` API must behave the same in VS Code and Theia.
[The doc says](https://github.com/eclipse-theia/theia/blob/c4f324ada78e8074c401fa52a0a655bbb64de017/packages/plugin/src/theia.d.ts#L5041-L5047) it must return `undefined` when no folder is open.

I've just filed a Theia bug https://github.com/eclipse-theia/theia/issues/8640
Working on fixing it. I'll let you know when it's fixed.

Reply:
@rgrunber I've fixed the bug in Theia https://github.com/eclipse-theia/theia/pull/8641 - it works identically to VS Code.